oonimkall/runner.go

```diff
@@ -59,34 +59,31 @@ func (r *runner) hasUnsupportedSettings() (unsupported bool) {
 		sadly("Options.Backend: not supported")
 	}
 	if r.settings.Options.CABundlePath != "" {
-		sadly("Options.CABundlePath: not supported")
+		logger.Warn("Options.CABundlePath: not supported")
 	}
 	if r.settings.Options.GeoIPASNPath != "" {
-		sadly("Options.GeoIPASNPath: not supported")
+		logger.Warn("Options.GeoIPASNPath: not supported")
 	}
 	if r.settings.Options.GeoIPCountryPath != "" {
-		sadly("Options.GeoIPCountryPath: not supported")
-	}
-	if r.settings.Options.NoFileReport != nil {
-		sadly("Options.NoFileReport: not supported")
+		logger.Warn("Options.GeoIPCountryPath: not supported")
 	}
 	if r.settings.Options.ProbeASN != "" {
-		sadly("Options.ProbeASN: not supported")
+		logger.Warn("Options.ProbeASN: not supported")
 	}
 	if r.settings.Options.ProbeCC != "" {
-		sadly("Options.ProbeCC: not supported")
+		logger.Warn("Options.ProbeCC: not supported")
 	}
 	if r.settings.Options.ProbeIP != "" {
-		sadly("Options.ProbeIP: not supported")
+		logger.Warn("Options.ProbeIP: not supported")
 	}
 	if r.settings.Options.ProbeNetworkName != "" {
-		sadly("Options.ProbeNetworkName: not supported")
+		logger.Warn("Options.ProbeNetworkName: not supported")
 	}
-	if r.settings.Options.RandomizeInput != nil {
+	if r.settings.Options.RandomizeInput != false {
 		sadly("Options.RandomizeInput: not supported")
 	}
-	if r.settings.OutputFilePath != "" {
-		sadly("OutputFilePath: not supported")
+	if r.settings.OutputFilepath != "" && r.settings.Options.NoFileReport == false {
+		sadly("OutputFilepath && !NoFileReport: not supported")
```

Comment:
These two cases have been conflated together because this seems to me the best way to guarantee a backward compatible semantics. The OONI app is setting `NoFileReport` to `true`. Historically MK has used an empty output file name to mean that MK would choose a name (what a terrible choice... 🤦‍♀🤦‍♀🤦‍♀🤦‍♀🤦‍♀🤦‍♀🤦‍♀).

The only aspect that makes me a little sorry here is that `NoFileReport` defaults to `false`, so a user integrating this library needs to set this field explicitly to `true` for the test to work. Anyway, it probably makes sense in the future to support the `OutputFilepath` option by default, but it mainly does not make sense to work on this now, since the OONI app manages the file.

As an historical note, the reason why we have chosen at a later stage to avoid writing on disk was because MK was supposed being an engine, therefore the app should have been responsible of the file system. However, it's not always that easy, and both probe-engine and MK manage a bunch of files, so maybe writing the result makes sense.

What we can actually do is to change the name of the field like `output_file_path` and give it the correct semantics, i.e., that we are going to write the file if the string is not empty.